MergeSharpWebAPI/Models/Graph.cs

```diff
@@ -147,6 +147,11 @@ public override void ApplySynchronizedUpdate(PropagationMessage receivedUpdate)
 
         GraphMsg received = (GraphMsg) receivedUpdate;
         this._canilleraGraph.ApplySynchronizedUpdate(received.cGraphMsg);
+
+        // _vertexInfo should contain only the vertices now in the graph

```

Comment:
ah this is done because nodes that get removed can't be added again because whenever a node is added they get a new guid, so there's no reason to keep them in the graph?

Reply:
No actually. So you're right a vertex should never be readded through the Add() method, but it can be readded to the _vertexInfo during some merges. Take the case when you concurrently add an edge to a vertex while I remove the vertex. When I removed the vertex, I removed its entry from _vertexInfo, but when I merge with your state, I'll need to readd the entry into _vertexInfo.

I do the delete because I want _vertexInfo to reflect what's in _canilleraGraph. It shouldn't be holding any information about vertices that aren't in _canilleraGraph. Otherwise _vertexInfo will contain a bunch of entries about vertices that are no longer in the graph and it might grow too big.

Reply:
right right gotchu sir